src/fides/api/service/privacy_request/request_runner_service.py

```diff
@@ -490,7 +490,11 @@ async def run_privacy_request(
             if not proceed:
                 return
 
-        if ConfigProxy(session).notifications.send_request_completion_notification:
+        if ConfigProxy(
+            session
+        ).notifications.send_request_completion_notification and not policy.get_rules_for_action(
+            action_type=ActionType.consent
```

Comment:
ok, this took a bit of a refresher for me, but i think i'm remembering now - we basically infer an action type for a given `PrivacyRequest` based on the `Policy`, which isn't _required_ to have only rules of a single action type, but practically we expect any `Policy` to have rules with different action types - right?

if that's the case, then this makes sense. if it's not the case, we'd have an edge case here where a hypothetical policy that does access + consent _wouldn't_ trigger a completion email even though it probably should. but that's just totally unrealistic, it's not like we'd actually have a policy with rules that have both of those action types, right!?

anyway, i think this is all good (if my assumptions are correct) - i just wonder when it'll be time for us to update some of our constraints/assumptions in our policy/rule/general DSR framework to be a bit more explicit and consistent about this expectation. i don't think there's a lot of functional risk at the moment, just makes this a bit harder to keep track of when reviewing code 😅 

curious to get your thoughts, this was just a little bit of a rabbit hole i went down before feeling totally good about approving 👍

Reply:
I hesitated a bit when I wrote this too but it was more around where this check should live. I thought about moving the check to `initiate_privacy_request_completion_email` and just skip the identity check for consent policies but that still takes us back to your concern of how to reliably determine the action type. As we have it now we have one policy per action type so we should be good for now.